src/api/common.ts

```diff
@@ -220,42 +219,17 @@ export function validate404(status: number): boolean {
     return validate2xx(status) || status === 404;
 }
 
-type TrackedKey = "trackedEntities" | "enrollments" | "events";
+export function parseTrackerPager(data: ParamTracker) {
+    return data.pager
+        ? data.pager
+        : {
+              page: data.page,
+              pageSize: data.pageSize,
+              total: data.total,
+              pageCount: data.pageCount,
+          };
+}
 
-export type InternalTrackerResponse<M extends D2ModelSchemaBase, Fields, Key extends TrackedKey> = {
-    instances?: SelectedPick<M, Fields>[];
-    pager: TrackedPager;
-    page: number;
-    pageSize: number;
-    // Only if requested with totalPages=true
-    total?: number;
-    pageCount?: boolean;
-} & Record<Key, SelectedPick<M, Fields>[] | undefined>;
-
-export type PublicTrackerResponse<M extends D2ModelSchemaBase, Fields> = {
-    instances: SelectedPick<M, Fields>[];
+interface ParamTracker extends TrackedPager {
     pager: TrackedPager;
```

Comment:
We set this prop as optional, that was the logic what motivated the `a || b` logic. And also `pager?: TrackedPager` in the 3 tracker responses from the api.